tests/forms_tests/field_tests/test_choicefield.py

```diff
@@ -87,3 +88,14 @@ def test_choicefield_disabled(self):
             '<select id="id_f" name="f" disabled><option value="J">John</option>'
             '<option value="P">Paul</option></select>'
         )
+
+    def test_choicefield_enumeration(self):
+        class FirstNames(models.TextChoices):
+            JOHN = 'J', 'John'
+            PAUL = 'P', 'Paul'
+
+        f = ChoiceField(choices=FirstNames.choices)
+        self.assertEqual('J', f.clean('J'))
```

Comment:
So we should probably use the enum here. Something like:
```suggestion
        self.assertEqual(FirstNames.JOHN.value, f.clean(FirstNames.JOHN))
```

Reply:
I think the current form is fine. I would leave it.